File: src/main/java/carelog/carelog/relation/web/dto/RelationResponse.java
package carelog.carelog.relation.web.dto;

import carelog.carelog.relation.domain.*;
import lombok.*;

import java.util.UUID;

@Getter
public class RelationResponse {
    private final UUID publicId;
    private final UUID managerPublicId;
    private final UUID customerPublicId;
    private final RelationStatus status;

    private RelationResponse(UUID publicId, UUID managerPublicId, UUID customerPublicId, RelationStatus status) {
        this.publicId = publicId;
        this.managerPublicId = managerPublicId;
        this.customerPublicId = customerPublicId;
        this.status = status;
    }

    public static RelationResponse from(Relation relation) {
        return new RelationResponse(
                relation.getPublicId(),
                relation.getManager().getPublicId(),
                relation.getCustomer().getPublicId(),
                relation.getStatus()
        );
    }
}

File: src/main/kotlin/carelog/carelog/relation/web/dto/RelationResponse.kt
package carelog.carelog.relation.web.dto

import carelog.carelog.relation.domain.Relation
import carelog.carelog.relation.domain.RelationStatus
import io.swagger.v3.oas.annotations.media.Schema
import java.util.UUID

data class RelationResponse(
    @Schema(description = "관계 공개 ID") val publicId: UUID,
    @Schema(description = "매니저 공개 ID") val managerPublicId: UUID,
    @Schema(description = "고객 공개 ID") val customerPublicId: UUID,
    @Schema(description = "고객 이름") val customerName: String,
    @Schema(description = "관계 상태") val status: RelationStatus,
) {
    companion object {
        fun from(relation: Relation) = RelationResponse(
            publicId = relation.publicId,
            managerPublicId = relation.manager.publicId,
            customerPublicId = relation.customer.publicId,
            customerName = relation.customer.name,
            status = relation.status,
        )
    }
}
